src/OddJobs/Job.hs

```diff
@@ -340,6 +343,10 @@ cancelJobIO :: Connection -> TableName -> JobId -> IO (Maybe Job)
 cancelJobIO conn tname jid =
   updateJobHelper tname conn (Failed, [Queued, Retry], Nothing, jid)
 
+killJobIO :: Connection -> TableName -> JobId -> IO (Maybe Job)
+killJobIO conn tname jid =
+  updateJobHelper tname conn (Failed, [Locked], Nothing, jid)
```

Comment:
> Poll for jobs whose status has changed to canceled recently and cancel them if they are running

So if I understand correctly, you want to make the job runner poll to cancel any job which is running? My reading of the ticket was that this would use something like [`cancel`](https://hackage.haskell.org/package/async-2.2.3/docs/Control-Concurrent-Async.html#v:cancel) to instead force the job to stop. Is there a reason to do it via polling

Reply:
The web-ui and the jobRunner are decoupled, so there is no way to actually kill a thread from the ui. And also I think the polling approach scales better for more than one jobRunner.

Reply:
Makes sense, thanks for the explanation.